test/agent-policy-tests/test-update-utils/test-integration-update.sh

```diff
@@ -0,0 +1,510 @@
+#!/usr/bin/env bash
```

Comment:
I'm not sure we should have a separate script to execute Terraform outside the existing test processes. Can you explain the intent of this script?

Reply:
Sure! Since the current integration test infrastructure does not check to see if the module handles updates correctly, this script is responsible for verifying that the update behavior is correct.

Reply:
Got it, so this is specifically to test updates. IMO we could do this with the existing integration framework without running Terraform ourselves. Particularly:

1. Add a fixture with parameterized inputs.
2. Run `kitchen_do converge` on the fixture
3. Run `kitchen_do verify` on the fixture (with proper gcloud checks in the tests)
4. Add/update a terraform.tfvars for the fixture
4. Run `kitchen_do converge` again
5. Run `kitchen_do verify` again.

Thoughts?

Reply:
Since I'm going to take a look at making a `task_helper_functions.sh` anyway, I'll see if this fits in with the structure, and I'll update this thread when I figure out if it makes sense!

Reply:
You'd probably add this as a custom function `test_integration_update`, you could also override the default `run_integration_tests` [function](https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/blob/master/infra/build/developer-tools/build/scripts/task_helper_functions.sh#L496).

Reply:
Sounds good; I ended up writing a custom function + adding a target in the makefile that runs the function with the handwritten integration tests.